gpiozero/i2c_devices.py

```diff
@@ -46,4 +46,346 @@ def __repr__(self):
+
+    GAIN_1_40MV = 0  # Maximum shunt voltage 40mV
+    GAIN_2_80MV = 1  # Maximum shunt voltage 80mV
+    GAIN_4_160MV = 2  # Maximum shunt voltage 160mV
+    GAIN_8_320MV = 3  # Maximum shunt voltage 320mV
+    GAIN_AUTO = -1  # Determine gain automatically
+
+    ADC_9BIT = 0  # 9-bit conversion time  84us.
+    ADC_10BIT = 1  # 10-bit conversion time 148us.
+    ADC_11BIT = 2  # 11-bit conversion time 2766us.
+    ADC_12BIT = 3  # 12-bit conversion time 532us.
+    ADC_2SAMP = 9  # 2 samples at 12-bit, conversion time 1.06ms.
+    ADC_4SAMP = 10  # 4 samples at 12-bit, conversion time 2.13ms.
+    ADC_8SAMP = 11  # 8 samples at 12-bit, conversion time 4.26ms.
+    ADC_16SAMP = 12  # 16 samples at 12-bit,conversion time 8.51ms
+    ADC_32SAMP = 13  # 32 samples at 12-bit, conversion time 17.02ms.
+    ADC_64SAMP = 14  # 64 samples at 12-bit, conversion time 34.05ms.
+    ADC_128SAMP = 15  # 128 samples at 12-bit, conversion time 68.10ms.
+
+    __ADDRESS = 0x40
+
+    __REG_CONFIG = 0x00
+    __REG_SHUNTVOLTAGE = 0x01
+    __REG_BUSVOLTAGE = 0x02
+    __REG_POWER = 0x03
+    __REG_CURRENT = 0x04
+    __REG_CALIBRATION = 0x05
+
+    __RST = 15
+    __BRNG = 13
+    __PG1 = 12
+    __PG0 = 11
+    __BADC4 = 10
+    __BADC3 = 9
+    __BADC2 = 8
+    __BADC1 = 7
+    __SADC4 = 6
+    __SADC3 = 5
+    __SADC2 = 4
+    __SADC1 = 3
+    __MODE3 = 2
+    __MODE2 = 1
+    __MODE1 = 0
+
+    __OVF = 1
+    __CNVR = 2
+
+    __BUS_RANGE = [16, 32]
+    __GAIN_VOLTS = [0.04, 0.08, 0.16, 0.32]
+
+    __CONT_SH_BUS = 7
+
+    __SHUNT_MILLIVOLTS_LSB = 0.01  # 10uV
+    __BUS_MILLIVOLTS_LSB = 4  # 4mV
+    __CALIBRATION_FACTOR = 0.04096
+    __MAX_CALIBRATION_VALUE = 0xFFFE  # Max value supported (65534 decimal)
+    # In the spec (p17) the current LSB factor for the minimum LSB is
+    # documented as 32767, but a larger value (100.1% of 32767) is used
+    # to guarantee that current overflow can always be detected.
+    __CURRENT_LSB_FACTOR = 32800
+
+    def __init__(self, shunt_ohms, max_expected_amps=None, **i2c_args):
+        """Construct the class.
+
+        Pass in the resistance of the shunt resistor and the maximum expected
+        current flowing through it in your system.
+
+        Arguments:
+        shunt_ohms -- value of shunt resistor in Ohms (mandatory).
+        max_expected_amps -- the maximum expected current in Amps (optional).
+        """
+        super().__init__(**i2c_args)
+        self._shunt_ohms = shunt_ohms
+        self._max_expected_amps = max_expected_amps
+        self._min_device_current_lsb = self._calculate_min_current_lsb()
+        self._gain = None
+        self._auto_gain_enabled = False
+        self._voltage_range = None
+        self._current_lsb = None
+        self._power_lsb = None
+
+    def configure(self, voltage_range=RANGE_32V, gain=GAIN_AUTO,
+                  bus_adc=ADC_12BIT, shunt_adc=ADC_12BIT):
+        """Configure and calibrate how the INA219 will take measurements.
```

Comment:
Feels like this should be part of the `__init__` method, rather than being a separate `configure` method? :thinking:

Reply:
Like I said, this was completley stolen from [chrisb2's pi_ina219](https://github.com/chrisb2/pi_ina219).
I can add a call to `configure` to `__init__`, but it might be left out on purpose. If you want to use the device with the same settings as it already has (without knowing or caring what they are), you can use it without calling the `configure` method.

Reply:
Again this is a call for @bennuttall or @waveform80 to make, but IIRC GPIO Zero tries to be much "higher level" / "easier to use" than other typically lower-level libraries.